src/DotRush.Debugging.NetCore/Extensions/NUnitFilterHack.cs

```diff
@@ -48,8 +77,112 @@ private static string UpdateNUnitRunSettings(string runSettings, string whereFil
-        var assemblies = Directory.GetFiles(assemblyDirectory, "*.dll", SearchOption.TopDirectoryOnly);
-        return assemblies.Any(a => Path.GetFileName(a).Contains("NUnit", StringComparison.OrdinalIgnoreCase));
+        try {
+            if (Path.GetFileName(assemblyPath).Contains("NUnit", StringComparison.OrdinalIgnoreCase))
+                return true;
+
+            var assemblyDirectory = Path.GetDirectoryName(assemblyPath);
+            if (string.IsNullOrEmpty(assemblyDirectory) || !Directory.Exists(assemblyDirectory))
+                return false;
+
+            var assemblies = Directory.GetFiles(assemblyDirectory, "*.dll", SearchOption.TopDirectoryOnly);
+            return assemblies.Any(a => Path.GetFileName(a).Contains("NUnit", StringComparison.OrdinalIgnoreCase));
+        }
+        catch (Exception) {
+            return false;
+        }
+    }
+
+    private static string? ExtractTestCaseFilterFromRunSettings(string? runSettings) {
+        if (string.IsNullOrEmpty(runSettings))
+            return null;
+
+        try {
+            var xdoc = System.Xml.Linq.XDocument.Parse(runSettings);
+            var filter = xdoc.Root?
+                .Element("RunConfiguration")?
+                .Element("TestCaseFilter")?
+                .Value?
+                .Trim();
+            currentClassLogger.Debug($"ExtractTestCaseFilterFromRunSettings: '{filter}'");
+            return filter;
+        } catch (Exception ex) {
+            currentClassLogger.Debug($"ExtractTestCaseFilterFromRunSettings failed: {ex.Message}");
+            return null;
+        }
+    }
+
+    public static string? TranslateRunSettingsCategoryFilter(string filter) {
+        if (string.IsNullOrWhiteSpace(filter))
+            return null;
+
+        currentClassLogger.Debug($"TranslateRunSettingsCategoryFilter input: '{filter}'");
+        filter = filter.Replace(" ", string.Empty);
+
+        string? Parse(string expr) {
+            if (expr.Length == 0)
+                return null;
+
+            if (expr[0] == '(' && expr[^1] == ')' && MatchingParens(expr))
+                return Parse(expr.Substring(1, expr.Length - 2));
+
+            var orParts = SplitTop(expr, '|');
+            if (orParts.Count > 1) {
+                var parsedParts = orParts.Select(p => Parse(p) ?? string.Empty).Where(p => !string.IsNullOrEmpty(p));
+                var result = string.Join(" or ", parsedParts);
+                return "(" + result + ")";
+            }
+
+            var andParts = SplitTop(expr, '&');
+            if (andParts.Count > 1) {
+                var parsedParts = andParts.Select(p => Parse(p) ?? string.Empty).Where(p => !string.IsNullOrEmpty(p));
+                var result = string.Join(" and ", parsedParts);
+                return "(" + result + ")";
+            }
+
+            if (expr.StartsWith("Category=", StringComparison.OrdinalIgnoreCase))
+                return "cat==" + expr.Substring("Category=".Length);
```

Comment:
Use span-based 'string.Concat' and 'AsSpan' instead of 'Substring' [CA1845](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1845)

```suggestion
                return string.Concat("cat==", expr.AsSpan("Category=".Length));
```